### PR TITLE
[polkit] Update Debian patches, drop patches already included. JB#56199

### PR DESCRIPTION
**Pull request by Thaodan**

### CODE REVIEW

File: rpm/polkit.spec [hunk 1]
```diff
@@ -106,15 +135,18 @@ install -D -m 644 %{SOURCE1} %{buildroot}%{_unitdir}/dbus-org.freedesktop.Policy
...
 %dir %{_datadir}/polkit-1/actions
 %{_datadir}/polkit-1/actions/org.freedesktop.policykit.policy
-%config %{_sysconfdir}/dbus-1/system.d/org.freedesktop.PolicyKit1.conf
+%config %{_datadir}/dbus-1/system.d/org.freedesktop.PolicyKit1.conf
 %config %{_sysconfdir}/pam.d/polkit-1
 %config %{_sysconfdir}/polkit-1
-%{_bindir}/*
+%{_bindir}/pkaction
+%{_bindir}/pkcheck
+%{_bindir}/pk-example-frobnicate
```
Comment (pvuorela) [inline rpm/polkit.spec]: This doesn't look like a thing that should be installed to devices.

Comment (pvuorela) [reply rpm/polkit.spec]: And looks like included already below in devel.

File: rpm/polkit.spec [hunk 2]
```diff
@@ -6,49 +6,79 @@ License:    LGPLv2+
...
+Patch0054: patches/debian/01_pam_polkit.patch
+Patch0055: patches/debian/02_gettext.patch
+Patch0056: patches/debian/05_revert-admin-identities-unix-group-wheel.patch
+Patch0057: patches/debian/06_systemd-service.patch
+Patch0058: patches/debian/10_build-against-libsystemd.patch
+Patch0059: patches/debian/Move-D-Bus-policy-file-to-usr-share-dbus-1-system.d.patch
+Patch0060: patches/debian/Statically-link-libpolkit-backend1-into-polkitd.patch
+Patch0061: patches/debian/Remove-example-null-backend.patch
+Patch0062: patches/debian/CVE-2021-3560.patch
+Patch0063: patches/0003-Support-for-annotation-identity-group-check.patch
```
Comment (Tomin1) [inline rpm/polkit.spec:74]: Why these three are backwards? 0003 first and then 0002 and 0001.

File: rpm/genpatchlist.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    echo Patch${zeros}${num}: patches/debian/"$patch"
+    export num=$((num+1))
+    if [ $((num/100)) -eq 1 ] ; then
+        zeros=0
+    elif [ $((num/10)) -eq 1 ] ; then
+        zeros=00
+    fi
+
+    # read patching order from debian quilt file
+done < rpm/patches/debian/series
```
Comment (Tomin1) [inline rpm/genpatchlist.sh:20]: I don't see this file listed in spec. All files under rpm/ directory must be listed as a patch or source file to get through CI.

File: rpm/polkit.spec (hunk 2 above)
Comment (Thaodan) [reply rpm/polkit.spec:74]: Because of Debian patch ordering, it doesn't apply properly in a different order.

Comment (Tomin1) [reply rpm/polkit.spec:74]: Ok, just seemed confusing to me.

File: rpm/polkit.spec [hunk 4]
```diff
@@ -5,50 +5,79 @@ Release:    1
...
-Patch30:    patches/0.113/README-Note-to-send-security-reports-via-DBus-s-mech.patch
-Patch31:    patches/master/Fix-multi-line-pam-text-info.patch
-Patch32:    patches/CVE-2018-1116.patch
-Patch33:    patches/0001-dbus-Use-systemd-service.patch
-Patch34:    patches/0002-build-Disable-gtk-doc-support.patch
-Patch35:    patches/0003-Support-for-annotation-identity-group-check.patch
-Patch36:    patches/0004-Fix-build-with-systemd-v233.patch
+Source1:    polkit.service
+Source2:    genpatchlist.sh
+Source3:    rpm/patches/debian/series
```
Comment (Tomin1) [inline rpm/polkit.spec]: I think you are not supposed to include `rpm/` directory itself in source file paths.

File: rpm/polkit.spec (hunk 2 above)
Comment (pvuorela) [reply rpm/polkit.spec:74]: But these are non-debian patches and used to be in numerical order? If we want the order changed, could then also update the file name numbering?

Comment (Thaodan) [reply rpm/polkit.spec:74]: > But these are non-debian patches and used to be in numerical order? If we want the order changed, could then also update the file name numbering?
Fixed that.

File: rpm/polkit.spec [hunk 5]
```diff
@@ -5,50 +5,79 @@ Release:    1
...
+Patch0055: patches/debian/06_systemd-service.patch
+Patch0056: patches/debian/10_build-against-libsystemd.patch
+Patch0057: patches/debian/Move-D-Bus-policy-file-to-usr-share-dbus-1-system.d.patch
+Patch0058: patches/debian/Statically-link-libpolkit-backend1-into-polkitd.patch
+Patch0059: patches/debian/Remove-example-null-backend.patch
+Patch0060: patches/debian/CVE-2021-3560.patch
+Patch0061: patches/0001-dbus-Use-systemd-service.patch
+Patch0062: patches/0002-build-Disable-gtk-doc-support.patch
+Patch0063: patches/0003-Support-for-annotation-identity-group-check.patch
+equires:   dbus
```
Comment (mkosola) [inline rpm/polkit.spec]: Should be Requires

**Review by mkosola — APPROVED** [review]

**Pull request merged**